src/Verlite.Core/VersionCalculator.cs

```diff
@@ -25,9 +25,12 @@ public static SemVer Bump(SemVer version, VersionCalculationOptions options, int
 				throw new ArgumentOutOfRangeException(nameof(height), height, "Must be greater than zero.");
 
 			SemVer ret = version;
-			ret.Prerelease ??= options.DefaultPrereleasePhase;
-			ret.Prerelease += $".{options.PrereleaseBaseHeight + (height - 1)}";
-			ret.BuildMetadata = options.BuildMetadata;
+			if (options.AutoIncrement != VersionPart.None)
+			{
+				ret.Prerelease ??= options.DefaultPrereleasePhase;
+				ret.BuildMetadata = options.BuildMetadata;
+				ret.Prerelease += $".{options.PrereleaseBaseHeight + (height - 1)}";
+			}
```

Comment:
I think rather than nesting it, guard it like so:

```patch
		public static SemVer Bump(SemVer version, VersionCalculationOptions options, int height)
		{
			if (height < 1)
				throw new ArgumentOutOfRangeException(nameof(height), height, "Must be greater than zero.");

+			if (options.AutoIncrement == VersionPart.None)
+				return version;
+
			SemVer ret = version;
			ret.Prerelease ??= options.DefaultPrereleasePhase;
			ret.Prerelease += $".{options.PrereleaseBaseHeight + (height - 1)}";
			ret.BuildMetadata = options.BuildMetadata;

			return ret;
		}
```